src/parser/parser.cpp

```diff
@@ -166,8 +166,57 @@ bool Parser::WriteParseTree(const std::string &filepath) {
     ofs << parse_tree_.dump(INDENTATION) << '\n';
     return true;
   }
-
   return false;
 }
 
+/**
+ * Supportive function for Parser::FormattedJSON
+ * @param value
+ * @return { text: { name: "value" } }
+ */
+json GetTextNode(const std::string &value) {
+  json j;
+  j["text"]["name"] = value;
+  return j;
+}
+
+/**
+ * Supportive function for Parser::FormattedJSON
+ * @param value
+ * @return the current key of the JSON head
+ */
+std::string GetHeadKey(const json &j) {
```

Comment:
this is not used. remove if not required.

Reply:
yes, resolved